src/main/java/com/n1analytics/paillier/util/BigIntegerUtil.java

```diff
@@ -87,6 +88,24 @@ public static BigInteger modPow(BigInteger base, BigInteger exponent, BigInteger
       return base.modPow(exponent, modulus);
```

Comment:
If `! USE_GMP`, then we fall back to `java.math.BigInteger::modPow` which is *not* secure, making this function's name a misnomer.